metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/NetworkFunctions.java

```diff
@@ -78,13 +78,16 @@ public Object apply(List<Object> list) {
 
     @Override
     public Object apply(List<Object> objects) {
+      if(objects.isEmpty()) {
+        return null;
+      }
       Object dnObj = objects.get(0);
       InternetDomainName idn = toDomainName(dnObj);
       if(idn != null) {
         String dn = dnObj.toString();
         String tld = idn.publicSuffix().toString();
-        String suffix = Iterables.getFirst(Splitter.on(tld).split(dn), null);
-        if(suffix != null)
+        String suffix = dn.substring(0, dn.length() - tld.length());
+        if(suffix != null )
```

Comment:
Could you just drop this if-else?  Substring never returns null, so it can be dropped entirely.